frame/support/procedural/src/lib.rs

```diff
@@ -843,6 +870,35 @@ pub fn register_default_impl(attrs: TokenStream, tokens: TokenStream) -> TokenSt
 	}
 }
 
+/// The optional attribute `#[runtime_type]` can be attached to `RuntimeCall`, `RuntimeEvent`,
+/// `RuntimeOrigin` or `PalletInfo` in an impl statement that has `#[register_default_impl]`
+/// attached to indicate that this item is generated by `construct_runtime`.
+///
+/// Attaching this attribute to such an item ensures that the combined impl generated via
+/// [`#[derive_impl(..)]`](`macro@derive_impl`)  will use the correct type auto-generated by
+/// [`construct_runtime!`].
+///
+/// As an example, if you have an impl item `#[runtime_type] type RuntimeEvent = ();` in
+/// your impl statement, the combined impl will have `type RuntimeEvent = RuntimeEvent;` instead.
```

Comment:
Imo some proper docs on how this work would be nice. Maybe you could use `docify` as well to generate some good example.